pkg/container-collection/k8s.go

```diff
@@ -181,3 +192,66 @@ func (k *K8sClient) ListContainers() (arr []Container, err error) {
 	}
 	return arr, nil
 }
+
+func getSocketPathFromKubelet(clientset *kubernetes.Clientset, nodeName string) (string, error) {
+	// In case there's an environment variable, it overrides the detection
+	envsps := []string{"INSPEKTOR_GADGET_DOCKER_SOCKETPATH", "INSPEKTOR_GADGET_CONTAINERD_SOCKETPATH", "INSPEKTOR_GADGET_CRIO_SOCKETPATH", "INSPEKTOR_GADGET_PODMAN_SOCKETPATH"}
+
+	for _, envsp := range envsps {
+		if os.Getenv(envsp) != "" {
+			return "", nil
```

Comment:
We could, but we do it when we create the runtime client:
https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-utils/containerutils.go#L54

Reply:
I thought about it initially. The detection of the socket path is specific to k8s (Using the /configz endpoint in k8s). It is done using the k8s client which later creates a runtime client. The reasoning I added the env variables here is to make sure the env variables override the auto-detection if present. Do you still think it is better moving it to `NewContainerRuntimeClient()`?

Reply:
fwiw, in general, I think that if the user expects that if it explicitly provided configuration of some sort, it should take priority. How do you suggest we move forward defining this behavior?

Reply:
Interesting. I wasn't aware of that.
I needed this auto-detection capability when I wrote my own operators that used Kubernetes based enrichment like: https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-collection/options.go#L365

Saying this carefully, but I think we can remove these hard-coded values as we fallback to these values anyway in the runtime client code: https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-utils/runtime-client/interface.go#L25

Reply:
Do we really need to make the env variable have the highest precedence? If we get the path from Kubernetes, can't we be 100% sure it's the right one to use? Perhaps we can print a warning in that case indicating we're using the one we got from Kubernetes. 

> Saying this carefully, but I think we can remove these hard-coded values as we fallback to these values anyway in the runtime client code: https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-utils/runtime-client/interface.go#L25

I agree. With this autodetection mechanism in place, defining those won't be that important anymore.

Reply:
As a user, I would think it's "weird" that I "configured" IG to use a socket path and it ends up ignoring it.

Nonetheless, I'll add a debug print if the auto-detect logic is able to obtain the socket path.

Reply:
As for removing these values, do you want me to remove them as part of this PR or in a follow-up PR as @eiffel-fl suggested? I think a separate PR makes sense.

Reply:
> As a user, I would think it's "weird" that I "configured" IG to use a socket path and it ends up ignoring it.

We added the possibility to configure them because there was not an auto-detection mechanism in place, but now that we have this, I'd say the autodetected path should take precedence, in any case, if we're able to detect it, it's the right one to use, isn't it? 

(We can discuss all of this in a follow PR if you prefer)

> As for removing the values, do you want me to remove them as part of this PR or in a follow-up PR as @eiffel-fl suggested? I think a separate PR makes sense.

For sure we can handle this in another PR, but it's a bit weird to me that most of users (the ones that use the template to deploy it) won't benefit from this change after this PR is merged.

Reply:
Since the config endpoint isn't officially documented, I cannot guarantee its accuracy. It seems to be working properly in all environments I personally tested (k3s, minikube, microK8s and AKS) and given this endpoint has been around for a while, as discussed in this and the previous stale PR, I think it should be safe to use.

I'll remove the values from the templates as part of this PR.

Reply:
OK. I think I'm convinced the auto-detect should take precedence over the environment variables. I'll remove that part.